cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/BuildCommand.java

```diff
@@ -272,6 +300,12 @@ public void execute() {
                 CommandUtil.exitError(this.exitWhenFinish);
                 return;
             }
+
+            // If a custom target path is set, we modify the targetDir in the build options with the new custom one
+            if (this.targetDir != null) {
+                buildOptions.setTargetDir(this.targetDir.toString());
```

Comment:
We shouldn't introduce a setter to BuildOptions. This breaks the immutable model of Project API. The correct way is to use the BuildOptionsBuilder. Check the `constructBuildOptions` method in BuildCommand

Reply:
Removed the setter from BuildOptions. Added the condtion to constructBuildOptions() instead